src/plugins/robots/generic/simulator/noise_injector.cpp

```diff
@@ -0,0 +1,114 @@
+/**
+ * @file <argos3/plugins/robots/generic/noise_injector.cpp>
+ *
+ * @author John Harwell - <john.r.harwell@gmail.com>
+ */
+
+#include <argos3/plugins/robots/generic/simulator/noise_injector.h>
+
+#include "light_default_sensor.h"
+
+namespace argos {
+
+   /****************************************/
+   /****************************************/
+
+   void CNoiseInjector::Init(TConfigurationNode& t_tree) {
+      try {
+         /* parse noise model */
+         GetNodeAttribute(t_tree, "model", m_strModel);
+
+         if ("none" == m_strModel) {
```

Comment:
Put this at end:

if(first type of noise) ...
else if (second type of noise) ...
else if (not "none") error